proxyserver/objhandlers.go

```diff
@@ -194,7 +194,7 @@ func (server *ProxyServer) ObjectPutHandler(writer http.ResponseWriter, request
 			return
 		}
 	}
-	if request.Header.Get("Content-Type") == "" {
+	if request.Header.Get("Content-Type") == "" || common.LooksTrue(request.Header.Get("X-Detect-Content-Type")) {
```

Comment:
It's to override the Content-Type your HTTP library may be sending whether you want it to or not.